Sources/Common/Concurrency/TaskExtension.swift

```diff
@@ -42,6 +42,29 @@ public extension Task where Success == Never, Failure == Error {
             }
         }
     }
+
+    static func periodic(delay: TimeInterval? = nil,
+                         interval: TimeInterval,
+                         operation: @escaping @Sendable () async throws -> Void,
+                         cancellationHandler: (@Sendable () async -> Void)? = nil) -> Task {
+
+        Task {
+            do {
+                if let delay {
+                    try await Task<Never, Never>.sleep(interval: delay)
```

Comment:
Should be able to simplify this down to `Task.sleep` I believe.

Reply:
I'll keep it because of this, but not sure why that's even needed. 🤷 

<img width="739" alt="Screenshot 2024-07-30 at 11 57 48 AM" src="https://github.com/user-attachments/assets/b8b56f1c-b7d4-41e5-b604-28e5c8b54bc4">